### PR TITLE
ROMFS: Fix quad +, cleanup quad x, remove trailing zeros

### PR DESCRIPTION
**Pull request by sfuhrer**
Reduces entropy and saves 1.5kB of flash.

### CODE REVIEW

**Comment by MaEtUgR** [comment]
We have to be careful with the numbering. I'm going through 👀

**Comment by MaEtUgR** [comment]
I checked through, corrected the small quads and added some savings.
image

File: ROMFS/px4fmu_common/init.d/airframes/4014_s500 [hunk 1]
```diff
@@ -12,20 +12,9 @@
...
 
 . ${R}etc/init.d/rc.mc_defaults
+. ${R}etc/init.d/rc.quad_x_default_geometry
 
 param set-default MC_ROLLRATE_P 0.18
 param set-default MC_PITCHRATE_P 0.18
 param set-default MC_ROLLRATE_I 0.15
 param set-default MC_PITCHRATE_I 0.15
-
-param set-default CA_ROTOR_COUNT 4
```
Comment (dagar) [inline ROMFS/px4fmu_common/init.d/airframes/4014_s500:21]: One thing to consider is that people will create their own airframe by forking the closest existing example and modifying it. Directly laying out all relevant parameters makes that as obvious as possible.

I'd like to get to the point where airframes are literally declarative files that we can validate statically.

For flash savings in general there are some things I believe we need to be pursuing fundamentally differently that will yield orders of magnitude larger savings for the way people actually want/need to use PX4.

Comment (MaEtUgR) [reply ROMFS/px4fmu_common/init.d/airframes/4014_s500:21]: So separate builds yes.

Is there a way to compress the ROMFS? That would anyways save better than the manual work.

**Comment by sfuhrer** [comment]
Thanks for the additions @MaEtUgR, almost 2kB savings now. LGTM.

File: ROMFS/px4fmu_common/init.d/airframes/4014_s500 (hunk 1 above)
Comment (sfuhrer) [reply ROMFS/px4fmu_common/init.d/airframes/4014_s500:21]: >One thing to consider is that people will create their own airframe by forking the closest existing example and modifying it. Directly laying out all relevant parameters makes that as obvious as possible.

We already had the rc.mc_defaults etc that we included, so this change here is just following up on that. And it really also reduces repetitive lines that add chances for error. For me that together with the flash savings overweights the small complexity increase. 

>For flash savings in general there are some things I believe we need to be pursuing fundamentally differently that will yield orders of magnitude larger savings for the way people actually want/need to use PX4.

Could you note them down eg in an Issue or somewhere? For us flash constraints are a big pain at the moment, so let's get the work rolling!

**Review by MaEtUgR — APPROVED** [review]
Rebased on main, removed a few trailing zeroes I forgot and checked again with bloaty.
image
@dagar Do you think this is a step back or is it an acceptable substituition?

**Comment by MaEtUgR** [comment]
@dagar I added a commit that adds the geometry back but keeps it consistent where there's no difference and adds a differrent comment where there is a difference. Is it more obvious now?

File: ROMFS/px4fmu_common/init.d/airframes/1002_standard_vtol.hil [hunk 2]
```diff
@@ -55,20 +54,23 @@ param set-default VT_TYPE 2
...
-param set-default CA_ROTOR2_KM -0.05
-param set-default CA_ROTOR3_PX -0.15
-param set-default CA_ROTOR3_PY 0.15
-param set-default CA_ROTOR3_KM -0.05
 param set-default CA_ROTOR4_AX 1.0
 param set-default CA_ROTOR4_AZ 0.0
 param set-default CA_ROTOR4_PX 0.2
 
+# Square quadrotor X PX4 numbering
+param set-default CA_ROTOR_COUNT 4
```
Comment (sfuhrer) [inline ROMFS/px4fmu_common/init.d/airframes/1002_standard_vtol.hil:62]: The count is 5, as it also has a pusher

Comment (sfuhrer) [inline ROMFS/px4fmu_common/init.d/airframes/1002_standard_vtol.hil:62]:
```suggestion
```

File: ROMFS/px4fmu_common/init.d/airframes/5001_quad_+ [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
-param set-default CA_ROTOR0_PY 0.15
-param set-default CA_ROTOR0_KM -0.05
-param set-default CA_ROTOR1_PX -0.15
-param set-default CA_ROTOR1_PY -0.15
-param set-default CA_ROTOR1_KM -0.05
-param set-default CA_ROTOR2_PX 0.15
-param set-default CA_ROTOR2_PY -0.15
-param set-default CA_ROTOR3_PX -0.15
-param set-default CA_ROTOR3_PY 0.15
+
```
Comment (MaEtUgR) [inline ROMFS/px4fmu_common/init.d/airframes/5001_quad_+:15]: Note: This was incorrectly also doing a Quad X. I corrected according to https://docs.px4.io/main/en/airframes/airframe_reference.html#quadrotor

**Review by sfuhrer — COMMENTED** [review]
@MaEtUgR you best drop my commit (7cd3cba90500e9ae1a8da89959e8cefb5544c0f5) completely, then we don't have to squash the commits on merging.

**Review by MaEtUgR — APPROVED** [review]

**Pull request merged**